### PR TITLE
feat: rework pointers usage

### PR DESCRIPTION
**Pull request by chumaknadya**

### CODE REVIEW

**Comment by chumaknadya** [comment]
initProposal - 224 bytes
 stateProposal - 144 bytes
 state - 129 bytes

File: app/pkg/protocol/state_proposal.go [hunk 1]
```diff
@@ -5,18 +5,18 @@ import (
...
 
 	"github.com/shopspring/decimal"
-	"github.com/statechannels/go-nitro/channel/state"
+	st "github.com/statechannels/go-nitro/channel/state"
 	"github.com/statechannels/go-nitro/types"
 )
 
 // StateProposal represents information about proposed state.
 type StateProposal struct {
-	state          *state.State
```
Comment (chumaknadya) [inline app/pkg/protocol/state_proposal.go:14]: state here is heavy one 129 byte(so lets use a pointer to state)

liabilityState is struct with map (delete pointer)

File: app/pkg/protocol/state_proposal.go [hunk 2]
```diff
@@ -5,18 +5,18 @@ import (
...
-	liabilityState *liability.LiabilityState
+	state          *st.State
+	liabilityState liability.LiabilityState
 }
 
 // NewStateProposal creates state proposal from state.
-func NewStateProposal(state *state.State) (*StateProposal, error) {
```
Comment (chumaknadya) [inline app/pkg/protocol/state_proposal.go:19]: go-critic linter suggest to have a pointer here (cause State is heavy)

File: app/pkg/protocol/conclude.go [hunk 3]
```diff
@@ -19,7 +19,7 @@ type concludeParams struct {
 }
 
 // buildConcludeParams builds conclude params for state channel finalization.
-func buildConcludeParams(s state.State, participantSignatures map[common.Address]state.Signature) (concludeParams, error) {
```
Comment (chumaknadya) [inline app/pkg/protocol/conclude.go:22]: go-critic linter suggestion

File: app/pkg/protocol/conclude.go [hunk 4]
```diff
@@ -61,7 +61,7 @@ func buildConcludeParams(s state.State, participantSignatures map[common.Address
 }
 
 // forceMoveSignatures forms signatures as IForceMoveSignature type.
-func forceMoveSignatures(s state.State, participantSignatures map[common.Address]state.Signature) []nitro.IForceMoveSignature {
```
Comment (chumaknadya) [inline app/pkg/protocol/conclude.go:64]: go-critic linter suggestion

File: app/pkg/protocol/channel.go [hunk 5]
```diff
@@ -188,7 +189,7 @@ func (channel *Channel) Conclude(p Participant, privateKey []byte, participantSi
 
 // CheckSignature returns true if signature is valid, existing in state channel participant list and
 // connected to the specific state, false otherwise.
-func (channel *Channel) CheckSignature(signature state.Signature, s state.State) (bool, error) {
```
Comment (chumaknadya) [inline app/pkg/protocol/channel.go:191]: go-critic suggestion

File: app/pkg/protocol/channel.go [hunk 6]
```diff
@@ -235,13 +236,13 @@ func (channel *Channel) StateIsFinal() bool {
 
 // signState adds a participant's signature to the newState.
 // An error is thrown if the signature is invalid.
-func (channel *Channel) signState(newState state.State, privateKey []byte) (state.Signature, error) {
```
Comment (chumaknadya) [inline app/pkg/protocol/channel.go:238]: go-critic suggestion

File: app/pkg/nitro/client.go [hunk 7]
```diff
@@ -30,7 +30,7 @@ type StateChannelContract interface {
 type Client struct {
 	Adjudicator StateChannelContract
 	ChainID     *big.Int
-	Eth         *ethclient.Client
```
Comment (chumaknadya) [inline app/pkg/nitro/client.go:33]: ``ethclient.Client`` consists of only 1 pointer

File: app/pkg/protocol/channel.go [hunk 8]
```diff
@@ -23,20 +23,20 @@ var (
...
 	lastState    *state.State
-	c            *channel.Channel
+	c            chl.Channel
 }
 
 // InitChannel opens channel with participant who was requested opening a channel.
-func InitChannel(initProposal InitProposal, participantIndex uint) (*Channel, error) {
-	c, err := channel.New(*initProposal.State, participantIndex)
+func InitChannel(initProposal *InitProposal, participantIndex uint) (*Channel, error) {
+	c, err := chl.New(*initProposal.State, participantIndex)
```
Comment (chumaknadya) [inline app/pkg/protocol/channel.go:33]: here as you see they return the copy, that's why I don't really see a reason to store a pointer in our side (its from go-nitro library)

you can see here that with New function they return copy and not a pointer 😅

**Review by lckmidnight — APPROVED** [review]

**Pull request merged**